Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -0,0 +1,365 @@
+
+#include "RGroupCore.h"
+
+namespace RDKit {
+
+namespace {
+// From answer 12 in
+// https://stackoverflow.com/questions/5279051/how-can-i-create-cartesian-product-of-vector-of-vectors
+// by anumi
+static std::vector<std::vector<int>> cartesianProduct(
+    const std::vector<std::vector<int>> &v) {
+  std::vector<std::vector<int>> s = {{}};
```

Comment:
The empty initializer list isn't necessary.

Reply:
I think it is- otherwise this line will do nothing (as s is empty) and the product will always be empty:

```cpp
 for (const auto &x : s) {
```

Reply:
Yeah, that's embarrassing. Sorry for the noise.